packages/library-legacy/src/utils/send-and-confirm-raw-transaction.ts

```diff
@@ -93,6 +93,14 @@ export async function sendAndConfirmRawTransaction(
   const status = (await confirmationPromise).value;
 
   if (status.err) {
+    if (signature != null) {
+      const transaction = await connection.getTransaction(signature);
+      if (transaction?.meta?.err) {
+        throw new Error(
+          `Transaction ${signature} failed: ${JSON.stringify(transaction.meta.logMessages, null, 2)}`,
+        );
+      }
+    }
```

Comment:
This should probably be in a try-catch, in case the fetching of the transaction also fails. Also, we should still be logging the error code.

Reply:
Good point. Added